functions/src/definitions/calculateSimilarity.ts

```diff
@@ -5,48 +5,59 @@ import {
   CollectionTypes,
 } from "./common/typesense/collectionOperations"
 import * as functions from "firebase-functions"
+import { getEmbedding } from "./common/machineLearningServer/operations"
+import { stripUrl, stripPhone } from "./common/utils"
 
 if (!admin.apps.length) {
   admin.initializeApp()
 }
 
 async function calculateSimilarity(
-  embedding: number[],
+  text: string,
   captionHash: string | null = null
 ) {
   //embed message to compare
-  const results = await vectorSearch(
-    embedding,
-    CollectionTypes.Instances,
-    1,
-    `captionHash:${captionHash ? captionHash : "__NULL__"}`
-  )
-  const nearestInstance = results?.[0]?.hits?.[0]
-  if (nearestInstance) {
-    const doc = nearestInstance.document
-    const path = doc.id.replace(/_/g, "/") //typesense id can't seem to take /
-    const similarityScore = 1 - nearestInstance.vector_distance
-    const db = admin.firestore()
-    const instanceRef = db.doc(path)
-    const instanceSnap = await instanceRef.get()
-    if (!instanceSnap.exists) {
-      functions.logger.warn("Path doesn't exist in database")
-      return {}
-    }
-    if (instanceSnap.get("text") != (doc.message ?? null)) {
-      functions.logger.warn("Text doesn't match")
-      return {}
-    }
-    return {
-      ref: instanceRef,
-      message: doc.message,
-      captionHash: instanceSnap.get("captionHash") ?? null,
-      score: Math.round(similarityScore * 100) / 100,
-      parent: instanceRef.parent.parent,
-    }
+
+  let similarity = {}
+  const embedding = await getEmbedding(text)
```

Comment:
I think we can get the embedding after checking if the remaining message is long enough? I assume there is a cost to asking for embeddings, so we can reduce unwanted api calls

Reply:
Think I still wanna get the embedding cos I wanna analyse how close the embeddings are in such cases!